spec/unit/util_spec.rb

```diff
@@ -233,7 +233,15 @@ def get_mode(file)
       # This is ugly, but I can't really think of a better way to do it without
       # letting it actually close fds, which seems risky
       (0..2).each {|n| IO.expects(:new).with(n).never}
-      (3..256).each {|n| IO.expects(:new).with(n).returns mock('io', :close) }
+
+      fds = Dir.entries("/proc/self/fd")
```

Comment:
This will fail on systems that don't have `/proc`, definitely windows, not sure about solaris. You may need to confine the example group to `:if => Puppet.features.posix?`